server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -766,9 +785,20 @@ private static void validateAliasOrIndex(String expression) {
 
         private static boolean aliasOrIndexExists(IndicesOptions options, Metadata metadata, String expression) {
```

Comment:
It looks like there's a lot of places in this class where `aliasOrIndex` could be changed to `indexAbstraction`. Maybe something for a follow-up PR.

Reply:
Yes, we should have a follow up pr that replaces `aliasOrIndex` with `indexAbstraction`.